### PR TITLE
[ramda] use keyof in R.keys

### PR DESCRIPTION
**Pull request by biern**
R.keys does not fail and just returns an empty list when given something other than an object so I guess the general definition should be kept intact.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
Waiting for approval from one of the many other authors.

This is a reasonably-observable breaking change, e.g.
```
const k = R.keys({a: 0});
k.push("b"); // new error
```

**Review by KiaraGrouwstra — APPROVED** [review]

**Comment by tycho01** [comment]
As Ryan noted, it's tough to cover all potential cases at once, and there is a trade-off involved here.
In this case though, I'm under the impression the benefits outweigh its downside.
For what it's worth, this particular library encouraging a coding style avoiding mutation may help make the push scenario less common here than cases that benefit from the more accurate return type.

**Review by moshensky — APPROVED** [review]

**Pull request merged**
